### PR TITLE
Add group

### PR DESCRIPTION
**Pull request by Bathande**

### CODE REVIEW

File: skietbaan_fe/src/actions/postActions.js [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
 		dispatch({
-			type: FETCH_POSTS,
-			payload: posts
+		  type: FETCH_POSTS,
+		  payload: newdata
+		});
+	  });
+  };
+  export const createGroups = usersadded => dispatch => {
+	fetch("http://localhost:63474/api/groups", {
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/postActions.js]: Do not commit localhost variables

File: skietbaan_fe/src/actions/postActions.js [hunk 2]
```diff
@@ -1,28 +1,80 @@
...
-		);
-};
+	  .then(function(response) {})
+	  .then(function(data) {})
+	  .catch(function(data) {});
+  };
+  
+  export const getname = name => {
+	//Return an action
+	console.log(name)
```
Comment (FredJEhlers) [inline skietbaan_fe/src/actions/postActions.js]: Console logs please remove

File: skietbaan_fe/src/components/AddGroup.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          <a href="#" class="fa fa-angle-left" />
+          <div className="center_label">
+            <b>GROUPS</b>
+          </div>
+        </div>
+        <div className="middle_bar">
+          <label className="name">
+            <b>Enter Group Name</b>
+          </label>
+          <br />
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/AddGroup.js]: Make use of css styles instead of `<br />`

File: skietbaan_fe/src/components/Groups.js [hunk 4]
```diff
@@ -1,117 +1,149 @@
...
+      filterText: ""
+    };
+    this.toggleHighlight = this.toggleHighlight.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.onBack = this.onBack.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
   componentWillMount() {
-    fetch('http://localhost:63474/api/user')
+    fetch("http://localhost:63474/api/user")
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: please make sure to get all the localhost instances

File: skietbaan_fe/src/components/Groups.js [hunk 5]
```diff
@@ -1,117 +1,149 @@
...
   toggleHighlight = event => {
-     if (this.state.posts[(event)].highlighted  === true) {
-       this.state.posts[(event)].highlighted=false;
-       {this.setState({count:this.state.count-1})}
-     }
-     else {
-       this.state.posts[(event)].highlighted=true;
-       {this.setState({count:this.state.count+1})}
-     }
+    console.log(event);
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: same goes for console.log

File: skietbaan_fe/src/components/Groups.js [hunk 6]
```diff
@@ -1,117 +1,149 @@
...
   render() {
-    const postItems = (   
-      <Table striped hover condensed>
-        <tbody >
-          {this.state.posts.map((post,index) => (
-            <tr key={post.id.toString()}  onClick={()=>this.toggleHighlight(index)} value={post.id} onChange={()=>this.onChange(post.id)}>
-              <td ><h5>{post.username} {index}</h5>
-                <h5>{post.email}</h5></td>
-            </tr>
+  //  console.log(this.props.groupNames.groupName)
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: GC

File: skietbaan_fe/src/components/Groups.js [hunk 7]
```diff
@@ -1,117 +1,149 @@
...
+    const postitems = (
+      <div className="check">
+        <ul class="list-group">
+
+          {this.state.posts.filter(
+            (post) => {
+              return (!this.state.filterText || post.username.toLowerCase().startsWith(this.state.filterText.toLowerCase()) || post.email.toLowerCase().startsWith(this.state.filterText.toLowerCase()))
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: Split this over mulitple lines

File: skietbaan_fe/src/components/Groups.js [hunk 8]
```diff
@@ -1,117 +1,149 @@
...
+      <main className="TheMain" >
+        <div className="TheNavBar">
+          <a href="#" class="fa fa-angle-left" onClick={this.onBack} />
+          <div className="center_label">
+            <b>Retro Rabbit</b>
+          </div>
+        </div>
+        <div className="BNavBar">
+          <input
+            className="theText"
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/Groups.js]: Hey please keep consistent with your naming scheme `CapitalCase`.

File: skietbaan_fe/src/components/GroupsName.js [hunk 9]
```diff
@@ -53,12 +53,12 @@ onSubmit(){
...
-    posts: state.posts.items,
-    newPost: state.posts.item,
-    groupName: state.groupName,
-  });
-const mapDispatchToProps = dispatch => ({
-    onGroupNameChange: (newGroupName) => dispatch({type: "UPDATE_GROUPNAME",payload:newGroupName})
-  });
-  export default connect(mapStateToProps,mapDispatchToProps)(GroupsName);
\ No newline at end of file
+// const mapStateToProps = state => ({
```
Comment (FredJEhlers) [inline skietbaan_fe/src/components/GroupsName.js]: GC
Empty line EOF

File: skietbaan_fe/src/actions/postActions.js [hunk 10]
```diff
@@ -1,28 +1,80 @@
...
+	NEW_POST,
+	UPDATENAME,
+	CREATEGROUP,
+	GETGROUP,
+	GETNAME
+  } from "./types";
+  
+  /** The method to feth the already available data for posts*/
+  export const fetchPosts = () => dispatch => {
+	fetch("http://localhost:63474/api/User")
```
Comment (CilliersN) [inline skietbaan_fe/src/actions/postActions.js]: change this fetch call

File: skietbaan_fe/src/components/Groups.js [hunk 11]
```diff
@@ -1,117 +1,149 @@
...
+      if (this.state.posts[i].highlighted === true) {
+        newArray.push(this.state.posts[i]);
+      }
+      delete this.state.posts[i].highlighted;
+      delete this.state.posts[i].id;
+    }
+    let request = {
+      newArray: this.state.newArray
+    };
+    console.log(request);
```
Comment (CilliersN) [inline skietbaan_fe/src/components/Groups.js]: remove console.logs

File: skietbaan_fe/src/components/add.css [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+  
+  .center_label {
+    display: flex;
+  
+  }
+  .center_label {
+    width: 800px;
+    margin-left: auto;
+    margin-right: auto;
+    width: 8em;
```
Comment (kimibeyl) [inline skietbaan_fe/src/components/add.css]: why duplicate width

File: skietbaan_fe/src/components/add.css [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+  .fa-angle-left {
+    color: black;
+    font-size: 48px;
+  }
+  
+  .nextpage {
+    position: absolute;
+    bottom: 0;
+    right: 0;
+    position: fixed;
```
Comment (kimibeyl) [inline skietbaan_fe/src/components/add.css]: why two positions

File: skietbaan_fe/src/actions/postActions.js [hunk 14]
```diff
@@ -1,28 +1,80 @@
...
+	return dispatch => {
+	  dispatch({
+		type: UPDATENAME,
+		payload: name
+	  });
+	};
+  };
+  
+  export const getGroup = () => dispatch => {
+	fetch("http://localhost:63474/api/User")
```
[the remark this answers is not in the document]
Comment (CilliersN) [reply skietbaan_fe/src/actions/postActions.js]: set localhost:63474 as a constant and add it in your env file.
pull ut from there

File: skietbaan_fe/src/actions/postActions.js (hunk 1 above)
Comment (Bathande) [reply skietbaan_fe/src/actions/postActions.js]: i have made changes

File: skietbaan_fe/src/actions/postActions.js [hunk 15]
```diff
@@ -1,28 +1,80 @@
...
-		},
-		body: JSON.stringify(postData)
+	  );
+  };
+  
+  {
+	/** the method to post new data from the PostForm */
+  }
+  export const createPost = users => dispatch => {
+	fetch("http://localhost:63474/api/groups/add", {
```
[the remark this answers is not in the document]
Comment (Bathande) [reply skietbaan_fe/src/actions/postActions.js]: i have made the change

File: skietbaan_fe/src/actions/postActions.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (Bathande) [reply skietbaan_fe/src/actions/postActions.js]: i made the change

File: skietbaan_fe/src/components/add.css (hunk 13 above)
Comment (Bathande) [reply skietbaan_fe/src/components/add.css]: i have took the redundant one out

File: skietbaan_fe/src/actions/postActions.js (hunk 10 above)
Comment (Bathande) [reply skietbaan_fe/src/actions/postActions.js]: i have made the change

File: skietbaan_fe/src/components/Groups.js (hunk 8 above)
Comment (Bathande) [reply skietbaan_fe/src/components/Groups.js]: made the chenges

File: skietbaan_fe/src/actions/postActions.js (hunk 10 above)
Comment (CilliersN) [reply skietbaan_fe/src/actions/postActions.js]: did you push?

File: skietbaan_fe/src/components/Groups.js [hunk 16]
```diff
@@ -1,117 +1,145 @@
...
     this.state = {
       posts: [],
-      newArray:[],
-      count:0,
-      black: ''  
-    }
-   this.toggleHighlight=this.toggleHighlight.bind(this);
-   this.handleOnClick=this.handleOnClick.bind(this);
-  } 
+
```
Comment (grootendorste) [inline skietbaan_fe/src/components/Groups.js]: Random line?

File: skietbaan_fe/src/components/Groups.js [hunk 17]
```diff
@@ -1,117 +1,145 @@
...
-       this.state.posts[(event)].highlighted=false;
-       {this.setState({count:this.state.count-1})}
-     }
-     else {
-       this.state.posts[(event)].highlighted=true;
-       {this.setState({count:this.state.count+1})}
-     }
+    if (this.state.posts[event].highlighted === true) {
+      this.state.posts[event].highlighted = false;
+      {
```
Comment (grootendorste) [inline skietbaan_fe/src/components/Groups.js]: Why random brackets?

**Pull request closed without merging**